### ISSUE TITLE
Metals snapshot not starting on java 8

### ISSUE THREAD

**Issue by ayoub-benali** [scalameta/metals#3402]
**Describe the bug**

Metals doesn't start when using latest snapshot `0.10.9+226-e28384c3-SNAPSHOT` on Java 8. 

From the error bellow it seems to be related to https://github.com/scalameta/metals/pull/2520

```
 Started: Metals version 0.10.10-SNAPSHOT in workspace '/home/ayoub/workspace/testproject' for client Sublime Text LSP 1.14.0.
Exception in thread "pool-4-thread-2" java.lang.UnsupportedClassVersionError: org/eclipse/text/edits/MalformedTreeException has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at scala.meta.internal.metals.MetalsLanguageServer.updateWorkspaceDirectory(MetalsLanguageServer.scala:509)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$initialize$2(MetalsLanguageServer.scala:799)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**To Reproduce**

1. Set Metals version to `0.10.9+226-e28384c3-SNAPSHOT` 
2. Open scala project
3. Open `.metals/metals.log`

**Expected behavior**

Metals to still run on Java 8 

**Installation:**

- Operating system: Linux
- Editor: Sublime
- Metals version: 0.10.9+226-e28384c3-SNAPSHOT

**Additional context**

```
java -version
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (build 1.8.0_312-8u312-b07-0ubuntu1~20.04-b07)
OpenJDK 64-Bit Server VM (build 25.312-b07, mixed mode)
```

**Comment by Arthurm1** [comment]
This commit should have fixed that...
https://github.com/scalameta/metals/commit/a4a681e5790ff358a81ff5dc949c3d3cfcfa254f

**Comment by ayoub-benali** [comment]
It seems that one of them at least isn't targeting java 8

**Comment by dos65** [comment]
As we are close to release I think the best option would be to move java formatter to a subproject.
This would allow us to fetch/run it dynamically in the same way as we work with mtags and do it only if jvm version is >= 11.

We can't drop java8 support as long as we keep scala 2.11 support.
We can also consider to replace eclipse formatter by some other one but I know nothing about java formatters state.

**Comment by ayoub-benali** [comment]
How about forking the formatter lib and publishing it for java 8 ?

**Comment by dos65** [comment]
@ayoub-benali I'm afraid it the most hardest option. It's not just a library - it's full eclipse platform or a part of it. 
Also, looks like a bit out of scalameta's scope :smile:

**Comment by Arthurm1** [comment]
@ayoub-benali Can you find out what version of `org.eclipse.text-XXXX.jar` you are using.  Ideally it should be `org.eclipse.text-3.9.0.jar`.  This should be listed in your `.bloop/metals.json` file if you're using Bloop.

Also - when you say you're using `0.10.9+226-e28384c3-SNAPSHOT`, is that a local snapshot you've published?  Or are metals snapshots published to a public repo somewhere - if so - can you tell me where?

**Comment by dos65** [comment]
@Arthurm1 each merge into main branch triggers snapshot release. [Here](https://scalameta.org/metals/docs/#latest-metals-server-versions) you can find the latest one.

**Comment by Arthurm1** [comment]
Ah - I see

Looks like the pom [here](https://oss.sonatype.org/content/repositories/public/org/scalameta/metals_2.12/0.10.9+228-ad3d4da0-SNAPSHOT/) doesn't take into account the `dependencyOverrides` i specified [here](https://github.com/scalameta/metals/pull/2520/commits/a4a681e5790ff358a81ff5dc949c3d3cfcfa254f).

In fact now that I've read the [manual](https://www.scala-sbt.org/1.x/docs/Library-Management.html#Overriding+a+version), it clearly states that `dependencyOverrides ` won't transfer to a published pom.xml.

I'll move the `dependencyOverrides` to pure dependencies and create a PR

**Comment by dos65** [comment]
Fixed by #3406

**Comment by ckipp01** [comment]
Closing since this was fixed in #3406

**Comment by dos65** [comment]
> Fixed by #3406

:facepalm:  I forgot to close it
